src/iceberg/table_metadata.h

```diff
@@ -76,6 +83,8 @@ struct ICEBERG_EXPORT TableMetadata {
   static constexpr int64_t kInvalidSequenceNumber = -1;
   static constexpr int64_t kInitialRowId = 0;
 
+  /// The location of the table metadata file
+  std::string metadata_file_location;
```

Comment:
I'm hesitant to introduce this field to the `TableMetadata` because `TableMetadata` aims to carry fields that only exist in the metadata json. Can we revert this?